op-node/rollup/derive/params.go

```diff
@@ -49,3 +51,19 @@ func (id ChannelID) TerminalString() string {
 func (id ChannelID) MarshalJSON() ([]byte, error) {
 	return json.Marshal(id.String())
 }
+
+func (id *ChannelID) UnmarshalJSON(b []byte) error {
+	s, err := strconv.Unquote(string(b))
+	if err != nil {
+		return err
+	}
+	h, err := hex.DecodeString(s)
+	if err != nil {
+		return err
+	}
+	if len(h) != ChannelIDLength {
+		return errors.New("invalid length")
+	}
+	copy(id[:], h)
+	return nil
+}
```

Comment:
Implementing MarshalText and UnmarshalText (see https://pkg.go.dev/encoding#TextMarshaler) may be simpler, since it avoids the json-quoting work. And the JSON golang code will handle it as a string then.

Reply:
I implemented this & it worked like a charm.